libs/agno/agno/vectordb/milvus/milvus.py

```diff
@@ -201,6 +278,35 @@ def upsert(self, documents: List[Document], filters: Optional[Dict[str, Any]] =
             )
             log_debug(f"Upserted document: {document.name} ({document.meta_data})")
 
+    async def async_upsert(self, documents: List[Document], filters: Optional[Dict[str, Any]] = None) -> None:
+        """Upsert documents asynchronously in parallel."""
+        print("--> async upsert")
+        log_debug(f"Upserting {len(documents)} documents asynchronously")
+
+        async def process_document(document):
+            document.embed(embedder=self.embedder)
+            cleaned_content = document.content.replace("\x00", "\ufffd")
+            doc_id = md5(cleaned_content.encode()).hexdigest()
+            data = {
+                "id": doc_id,
+                "vector": document.embedding,
+                "name": document.name,
+                "meta_data": document.meta_data,
+                "content": cleaned_content,
+                "usage": document.usage,
+            }
+            await self.async_client.upsert(
+                collection_name=self.collection,
+                data=data,
+            )
+            log_debug(f"Upserted document asynchronously: {document.name} ({document.meta_data})")
+            return data
+
+        # Process all documents in parallel
+        await asyncio.gather(*[process_document(doc) for doc in documents])
```

Comment:
No batch size here? I think actually lets get rid of batch size altogether? It should be fine to go full out with it